akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala

```diff
@@ -36,37 +38,25 @@ private object PoolSlot {
     final case class ConnectedEagerly(slotIx: Int) extends SlotEvent
   }
 
-  private val slotProcessorActorName = SeqActorName("SlotProcessor")
-
   /*
     Stream Setup
     ============
 
-    Request-   +-----------+              +-------------+              +-------------+     +------------+
-    Context    | Slot-     |  List[       |   flatten   |  Processor-  |   doubler   |     | SlotEvent- |  Response-
-    +--------->| Processor +------------->| (MapConcat) +------------->| (MapConcat) +---->| Split      +------------->
-               |           |  Processor-  |             |  Out         |             |     |            |  Context
-               +-----------+  Out]        +-------------+              +-------------+     +-----+------+
-                                                                                                 | RawSlotEvent
-                                                                                                 | (to Conductor
-                                                                                                 |  via slotEventMerge)
-                                                                                                 v
+    Request-   +-----------+              +-------------+              +------------+
+    Context    | Slot-     |  List[       |   flatten   |  Processor-  | SlotEvent- |  Response-
+    +--------->| Processor +------------->| (MapConcat) +------------->| Split      +------------->
+               |           |  Processor-  |             |  Out         |            |  Context
+               +-----------+  Out]        +-------------+              +-----+------+
+                                                                             | RawSlotEvent
+                                                                             | (to Conductor
+                                                                             |  via slotEventMerge)
+                                                                             v
    */
   def apply(slotIx: Int, connectionFlow: Flow[HttpRequest, HttpResponse, Any])(implicit system: ActorSystem, fm: Materializer): Graph[FanOutShape2[SlotCommand, ResponseContext, RawSlotEvent], Any] =
```

Comment:
As first step this can be simplified as

```scala
def apply(slotIx: Int, connectionFlow: Flow[HttpRequest, HttpResponse, Any])(implicit m: Materializer) = 
    GraphDSL.create() { implicit b ⇒
      import GraphDSL.Implicits._

      val log = ActorMaterializerHelper.downcast(m).logger
      val slotProcessor: FlowShape[SlotCommand, ProcessorOut] = b.add(new SlotProcessor(slotIx, connectionFlow, log))
      val partition = b.add(Partition[ProcessorOut](2, {
        case ResponseDelivery(r) ⇒ 0
        case r: RawSlotEvent ⇒ 1
      }))
      slotProcessor.out ~> partition.in

      val deliveries = partition.out(0).map(_.asInstanceOf[ResponseDelivery])
      val events = partition.out(1).map(_.asInstanceOf[RawSlotEvent])
      new FanOutShape2[SlotCommand, ResponseDelivery, RawSlotEvent](slotProcessor.in, deliveries.outlet, events.outlet)
    }
```